internal/config/migrate.go

```diff
@@ -0,0 +1,56 @@
+package config
+
+import (
+	"fmt"
+
+	ghConfig "github.com/cli/go-gh/v2/pkg/config"
+)
+
+//go:generate moq -rm -out migration_mock.go . Migration
+
+// Migration is the interace that config migrations must implement.
+//
+// Migrations will receive a copy of the config, and should modify that copy
+// as necessary. After migration has completed, the modified config contents
+// will be used.
+//
+// The calling code is expected to verify that the current version of the config
+// matches the PreVersion of the migration before calling Do, and will set the
+// config version to the PostVersion after the migration has completed successfully.
+type Migration interface {
+	// PreVersion is the required config version for this to be applied
+	PreVersion() string
+	// PostVersion is the config version that must be applied after migration
+	PostVersion() string
+	// Do is expected to apply any necessary changes to the config in place
+	Do(*ghConfig.Config) error
+}
+
+func Migrate(c *ghConfig.Config, m Migration) error {
+	// It is expected initially that there is no version key because we don't
+	// have one to begin with, so an error is expected.
+	version, _ := c.Get([]string{versionKey})
+
+	// If migration has already occured then do not attempt to migrate again.
+	if m.PostVersion() == version {
+		return nil
```

Comment:
Somewhere along the line this became untested, probably an easy and nice one to have.